pkg/prometheus/resource_selector.go

```diff
@@ -346,6 +346,11 @@ func validateScrapeClass(p monitoringv1.PrometheusInterface, sc *string) error {
 		if c.Name == *sc {
 			return nil
 		}
+		if c.Relabelings != nil {
+			if err := validateRelabelConfigs(p, c.Relabelings); err != nil {
+				return fmt.Errorf("scrapeClass %q relabelings: %w", *sc, err)
+			}
+		}
```

Comment:
Ideally it would need to happen before (e.g. at the beginning of the reconciliation) because it's a problem with the Prometheus/PrometheusAgent spec not with the monitoring object.

Reply:
Oh makes sensé but I could not find where to put the validation. Could you give me a hint ?

Reply:
I think I found a nice place in the `NewConfigGenerator` function. Can you check the new version?